### ISSUE TITLE
overrides paths don't support `./`

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#9953]
**Tell us about your environment**

* **ESLint Version:** v4.16.0
* **Node Version:** v8.9.1
* **npm Version:** v5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

mostly irrelevant except:
```json
"overrides": [
  {
    "files": ["./foo.js"],
    "rules": {
      "no-undef": 2,
    },
  },
],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
foo = 3;
```

```bash
eslint foo.js
```

**What did you expect to happen?**
A warning.

**What actually happened? Please include the actual, raw output from ESLint.**
No warning.

When I changed `./foo.js` to `foo.js`, I got a warning.

In other words, I think that overrides globs should be able to support leading `.` or `/`, like a normal node path + globs.

**Comment by platinumazure** [comment]
Definitely seems like a bug to me, but I don't know 100% how this is supposed to be designed. I know that we have some special-casing for dotfiles and I'm really hoping that isn't somehow causing this case.

**Comment by ljharb** [comment]
As mentioned in gitter by @not-an-aardvark, my expectation (that `./foo.js` and `foo.js` should be the same) is probably incorrect; but certainly if `foo.js` is a sibling to the config file, `./foo.js`, `foo.js`, and `/foo.js` should all target it the same.

**Comment by rwjblue** [comment]
I believe this is essentially a duplicate of https://github.com/eslint/eslint/issues/9740

**Comment by ljharb** [comment]
It might indeed be obviated by a solution to #9740, but I think it's a different enough problem that it should be its own issue.

**Comment by ljharb** [comment]
This is still an issue; please reopen it.
